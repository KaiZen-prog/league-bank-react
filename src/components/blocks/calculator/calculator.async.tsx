import {lazy} from 'react';

const CalculatorAsync = lazy(() => import('./index'));

export default CalculatorAsync;
