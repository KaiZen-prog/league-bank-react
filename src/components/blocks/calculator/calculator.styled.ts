import styled, {StyledComponentBase} from 'styled-components';
import {section} from '../../../theme/mixins';

interface StyledComponent extends StyledComponentBase<any, object> {}

const CalculatorSection: StyledComponent = styled.section`
  ${section()};
`;

export default CalculatorSection;
