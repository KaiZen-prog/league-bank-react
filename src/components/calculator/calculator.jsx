import React from "react";
import {CreditPurpose} from "../../const";
import withCalculator from "../../hocs/with-calculator/with-calculator";
import {divideNumberToSpace} from "../../utils/common";
import InputMask from "react-input-mask";
import PropTypes from "prop-types";
import {servicesSlides} from "../../mocks/mocks";

const Calculator = (props) => {
  const {
    state,
    onSelectOpen,
    onSelectClose,
    onPurposeChange,
    onInputFocus,
    onInputBlur,
    onInputChange,
    onCostChange,
    onInitialFeeChange,
    onTermChange,
    onInputRangeChange,
    onAdditionalChange,
    onCostChangeSign,
    onMakeRequest,
    onSubmit,
    onPopupClose,
    onRegApplicationChange,
    onChangePhone,
    requestNumber,
  } = props;

  const {
    step,
    purpose,
    isPurposeSelectOpened,
    paramsCredit,

    cost,
    initialFee,
    term,

    maternalCapital,
    casco,
    lifeInsurance,

    creditAmount,
    percent,
    monthlyPayment,
    requiredIncome
  } = state;

  const getRangeValuePosition = () => {
    let position = ((initialFee * 100 / cost) - paramsCredit.minInitialFee) * 100 / (100 - paramsCredit.minInitialFee);
    if (position < 0) position = 0;
    if (position > 100) position = 100;

    return position;
  }

  return (
      <section className="calculator">
        <a name="calculator"></a>
        <form action="#" className="calculator__form" onSubmit={onMakeRequest}>
          <h2 className="calculator__title">Кредитный калькулятор</h2>
          <div className="calculator__flex-container">
            <div className="calculator__container">
              <fieldset className="calculator__purpose">
                <h3 className="calculator__step">Шаг 1. Цель кредита</h3>
                <div
                    className={`calculator__purpose-select${isPurposeSelectOpened ? ' calculator__purpose-select--opened' : ''}`}
                    onClick={isPurposeSelectOpened ? onSelectClose : onSelectOpen}
                >
                  <span className="calculator__select-title">{CreditPurpose[purpose]}</span>
                  <ul className={`calculator__purpose-list${isPurposeSelectOpened ? '' : ' calculator__purpose-list--hidden'}`}>
                    <li className="calculator__purpose-item" id="mortgage" onClick={onPurposeChange}>Ипотечное кредитование</li>
                    <li className="calculator__purpose-item" id="car" onClick={onPurposeChange}>Автомобильное кредитование</li>
                  </ul>
                </div>
              </fieldset>
              {step >= 2 && (
                  <fieldset className="credit-calculator__params">
                    <h3 className="credit-calculator__step credit-calculator__step--params">Шаг 2. Введите параметры кредита</h3>
                    <label className="credit-calculator__label credit-calculator__label--cost">
                      <h4 className="credit-calculator__field-title">Стоимость {paramsCredit.type === `mortgage` ? `недвижимости` : `автомобиля`}</h4>

                      <span className="credit-calculator__minus-icon" id="minus" onClick={onCostChangeSign}></span>
                      <input
                          type="number"
                          className="credit-calculator__input"
                          name="cost"
                          min={paramsCredit.minCost}
                          max={paramsCredit.maxCost}
                          value={cost}
                          onBlur={onCostChange}
                          onChange={onInputChange}
                      />
                      <div className="credit-calculator__input credit-calculator__input--show" tabIndex="0" onFocus={onInputFocus}>{typeof cost === `string` ? cost : divideNumberToSpace(cost) + ` рублей`}</div>
                      <span className="credit-calculator__plus-icon" id="plus" onClick={onCostChangeSign}></span>

                      <p className="credit-calculator__help-text">От {divideNumberToSpace(paramsCredit.minCost)} &nbsp;до {divideNumberToSpace(paramsCredit.maxCost)} рублей</p>
                    </label>

                    <label className="credit-calculator__label credit-calculator__label--initial-fee">
                      <h4 className="credit-calculator__field-title">Первоначальный взнос</h4>

                      <input
                          type="number"
                          className="credit-calculator__input credit-calculator__input--initial-fee"
                          name="initialFee"
                          min={paramsCredit.minCost * paramsCredit.minInitialFee / 100}
                          max={paramsCredit.maxCost}
                          value={initialFee}
                          onBlur={onInitialFeeChange}
                          onChange={onInputChange}
                      />
                      <div className="credit-calculator__input credit-calculator__input--show credit-calculator__input--initial-fee" tabIndex="0" onFocus={onInputFocus}>{divideNumberToSpace(initialFee)} рублей</div>

                      <input
                          type="range"
                          className="credit-calculator__input-range"
                          name="initialFee"
                          min={paramsCredit.minInitialFee}
                          max="100"
                          step="5"
                          value={initialFee * 100 / cost}
                          onChange={onInputRangeChange}
                      />
                      <span
                          className="credit-calculator__range-value"
                          style={{
                            marginLeft: getRangeValuePosition() + `%`,
                            transform: `translateX(-${getRangeValuePosition() / 2}%)`
                          }}
                      >{isNaN(Math.floor(initialFee * 100 / cost)) ? 0 : Math.floor(initialFee * 100 / cost)}%</span>
                    </label>

                    <label className="credit-calculator__label credit-calculator__label--term">
                      <h4 className="credit-calculator__field-title">Срок кредитования</h4>

                      <input
                          type="number"
                          className="credit-calculator__input credit-calculator__input--term"
                          name="term"
                          min={paramsCredit.minTerm}
                          max={paramsCredit.maxTerm}
                          value={term}
                          onBlur={onTermChange}
                          onChange={onInputChange}
                      />
                      <div className="credit-calculator__input credit-calculator__input--show credit-calculator__input--term" tabIndex="0" onFocus={onInputFocus}>{term} лет</div>

                      <input
                          type="range"
                          className="credit-calculator__input-range credit-calculator__input-range--term"
                          name="term"
                          min={paramsCredit.minTerm}
                          max={paramsCredit.maxTerm}
                          step="1"
                          value={term}
                          onChange={onInputRangeChange}
                      />
                      <div className="credit-calculator__term-container">
                        <span className="credit-calculator__range-value">{paramsCredit.minTerm} {paramsCredit.minTerm === 1 ? `год` : `лет`}</span>
                        <span className="credit-calculator__range-value">{paramsCredit.maxTerm} лет</span>
                      </div>
                    </label>

                    {paramsCredit.maternalCapitalValue && (
                        <label className="credit-calculator__additional">
                          <input type="checkbox" name="maternalCapital" className="credit-calculator__input-checkbox visually-hidden" onChange={onAdditionalChange}/>
                          <span className="credit-calculator__checkbox-icon"></span>
                          Использовать материнский капитал
                        </label>
                    )}
                    {paramsCredit.additionalToCar && (
                        <>
                          <label className="credit-calculator__additional">
                            <input type="checkbox" name="casco" className="credit-calculator__input-checkbox visually-hidden" onChange={onAdditionalChange}/>
                            <span className="credit-calculator__checkbox-icon"></span>
                            Оформить КАСКО в нашем банке
                          </label>
                          <label className="credit-calculator__additional">
                            <input type="checkbox" name="lifeInsurance" className="credit-calculator__input-checkbox visually-hidden" onChange={onAdditionalChange}/>
                            <span className="credit-calculator__checkbox-icon"></span>
                            Оформить Страхование жизни в нашем банке
                          </label>
                        </>
                    )}
                  </fieldset>
              )}
            </div>
            {step >= 2 && (
                <>
                  {creditAmount >= paramsCredit.minCreditAmount && (
                      <div className="credit-calculator__offer">
                        <h3 className="credit-calculator__offer-title">Наше предложение</h3>
                        <ul className="credit-calculator__offer-list">
                          <li className="credit-calculator__offer-item">
                            <p className="credit-calculator__offer-value">{divideNumberToSpace(creditAmount)} рублей</p>
                            <p className="credit-calculator__offer-name">Сумма ипотеки</p>
                          </li>
                          <li className="credit-calculator__offer-item credit-calculator__offer-item--monthly-payment">
                            <p className="credit-calculator__offer-value">{divideNumberToSpace(monthlyPayment)} рублей</p>
                            <p className="credit-calculator__offer-name">Ежемесячный платеж</p>
                          </li>
                          <li className="credit-calculator__offer-item credit-calculator__offer-item--interest-rate">
                            <p className="credit-calculator__offer-value">{percent}%</p>
                            <p className="credit-calculator__offer-name">Процентная ставка</p>
                          </li>
                          <li className="credit-calculator__offer-item credit-calculator__offer-item--income">
                            <p className="credit-calculator__offer-value">{divideNumberToSpace(requiredIncome)} рублей</p>
                            <p className="credit-calculator__offer-name">Необходимый доход</p>
                          </li>
                        </ul>
                        <button type="submit" className="credit-calculator__submit-btn">Оформить заявку</button>
                      </div>
                  )}
                  {creditAmount < paramsCredit.minCreditAmount && (
                      <div className="credit-calculator__offer credit-calculator__offer--refusal">
                        <h3 className="credit-calculator__offer-title credit-calculator__offer-title--refusal">Наш банк не выдаёт ипотечные кредиты меньше {divideNumberToSpace(paramsCredit.minCreditAmount)} рублей.</h3>
                        <p className="credit-calculator__offer-name credit-calculator__offer-name--refusal">Попробуйте использовать другие параметры для расчёта.</p>
                      </div>
                  )}
                </>
            )}
          </div>
        </form>
        {step >= 3 && (
            <form action="#" className="credit-calculator__reg-application reg-application" onSubmit={onSubmit}>
              <h3 className="reg-application__title">Шаг 3. Оформление заявки</h3>
              <table className="reg-application__table">
                <tbody>
                <tr className="reg-application__param-field">
                  <td className="reg-application__field-name">Номер заявки</td>
                  <td className="reg-application__field-value">№ {(`0000` + requestNumber).slice(-4)}</td>
                </tr>
                <tr className="reg-application__param-field">
                  <td className="reg-application__field-name">Цель кредита</td>
                  <td className="reg-application__field-value">{purpose === `mortgage` ? `Ипотека` : `Автокредит`}</td>
                </tr>
                <tr className="reg-application__param-field">
                  <td className="reg-application__field-name">Стоимость {purpose === `mortgage` ? `недвижимости` : `автомобиля`}</td>
                  <td className="reg-application__field-value">{divideNumberToSpace(cost)} рублей</td>
                </tr>
                <tr className="reg-application__param-field">
                  <td className="reg-application__field-name">Первоначальный взнос</td>
                  <td className="reg-application__field-value">{divideNumberToSpace(initialFee)} рублей</td>
                </tr>
                <tr className="reg-application__param-field">
                  <td className="reg-application__field-name">Срок кредитования</td>
                  <td className="reg-application__field-value">{term} лет</td>
                </tr>
                </tbody>
              </table>
              <div className="reg-application__input-container">
                <input
                    type="text"
                    name="fullname"
                    className="reg-application__input reg-application__input--full-name"
                    placeholder="ФИО"
                    onChange={onRegApplicationChange}
                    onInvalid={(evt) => {
                      shakeEffect(evt.target);
                    }}
                    value={localStorage.getItem(`fullname`) !== null ? localStorage.getItem(`fullname`) : ``}
                    autoFocus
                    required/>
                <InputMask
                    mask="+7 (999) 999-9999"
                    maskChar=""
                    type="tel"
                    name="tel"
                    minLength={17}
                    className="reg-application__input reg-application__input--phone"
                    placeholder="Телефон"
                    onChange={onChangePhone}
                    onInvalid={(evt) => {
                      shakeEffect(evt.target);
                    }}
                    value={localStorage.getItem(`tel`) !== null ? localStorage.getItem(`tel`) : ``}
                    required
                />
                <input
                    type="email"
                    name="email"
                    className="reg-application__input"
                    placeholder="E-mail"
                    onChange={onRegApplicationChange}
                    onInvalid={(evt) => {
                      shakeEffect(evt.target);
                    }}
                    value={localStorage.getItem(`email`) !== null ? localStorage.getItem(`email`) : ``}
                    required/>
              </div>
              <button type="submit" className="reg-application__submit-btn">Отправить</button>
            </form>
        )}
        {step >= 4 && (
            <div className="popup-feedback" onClick={onPopupClose}>
              <div className="popup-feedback__container" onClick={(evt) => evt.stopPropagation()}>
                <button className="popup-feedback__close-btn" onClick={onPopupClose}></button>
                <h2 className="popup-feedback__title">Спасибо за обращение в наш банк.</h2>
                <p className="popup-feedback__content">Наш менеджер скоро свяжется с вами по указанному номеру телефона.</p>
              </div>
            </div>
        )}
      </section>
  );
};

Calculator.propTypes = {

};

Calculator.displayName = `Calculator`;

export default withCalculator(Calculator);
