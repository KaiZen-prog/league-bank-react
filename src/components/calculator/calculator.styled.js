import styled from 'styled-components';
import {css} from 'styled-components';
import theme from '../../theme/theme';
import {InputTypes} from '../../const';
import {backgroundImage, input, submitButton} from '../../theme/mixins';

import iconPurposeSelect from '../../img/icon-purpose-select.svg';
import iconPopupClose from '../../img/icon-close.svg';

const Calculator = styled.section`
  padding: 53px 20px 10px;

  margin: auto;

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    padding: 69px 42px 40px;
  }

  @media (min-width: ${theme.desktopWidthMinThreshold}) {
    padding: 96px 20px 40px 20px;

    max-width: 1210px;
    margin: auto;
  }
`;

Calculator.Title = styled.p`
  font-style: normal;
  font-weight: bold;
  font-size: 22px;
  line-height: 31px;

  margin-top: 0;
  margin-bottom: 22px;
  padding-left: 3px;

  color: ${theme.color.jaguar};

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    font-size: 32px;
    line-height: 45px;

    margin-bottom: 28px;
    padding-left: 3px;
  }

  @media (min-width: ${theme.desktopWidthMinThreshold}) {
    font-size: 41px;
    line-height: 57px;

    margin-bottom: 49px;
    padding-left: 0;
  }
`;

Calculator.FlexContainer = styled.div`
  @media (min-width: ${theme.desktopWidthMinThreshold}) {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
  }
`;

Calculator.Container = styled.div`
  @media (min-width: ${theme.desktopWidthMinThreshold}) {
    flex-grow: 1;
    max-width: 600px;
    margin-right: 60px;
  }
`;

Calculator.Purpose = styled.fieldset`
  border: none;

  margin: 0;
  padding: 0 3px 0;

  @media (min-width: ${theme.desktopWidthMinThreshold}) {
    padding: 0;
  }
`;

Calculator.PurposeSelect = styled.div`
  position: relative;

  width: 100%;

  border: 1px solid ${theme.color.jaguar};
  box-sizing: border-box;
  border-radius: 4px;

  &::before {
    top: 24px;
    right: 13px;

    ${backgroundImage(iconPurposeSelect, '18px', '11px')};

    cursor: pointer;
  }

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    &::before {
      right: 22px;
    }
  }

  ${(props) => {
    if(props.$isOpened) {
      return css`
        padding-bottom: 0;

        &::before {
          transform: rotate(180deg);
        }
      `;
    }
  }}
`;

Calculator.PurposeSelectTitle = styled.span`
  display: block;

  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 22px;

  color: ${theme.color.jaguar};

  padding: 19px 35px 17px 14px;

  cursor: pointer;

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    padding: 19px 50px 17px 23px;
  }
`;

Calculator.PurposeList = styled.ul`
  list-style: none;

  margin: 0;
  padding-left: 0;

  ${(props) => {
    if(props.$isClosed) {
      return css`
        display: none;
      `;
    }
  }}
`;

Calculator.PurposeItem = styled.li`
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;

  color: ${theme.color.jaguar};

  padding: 20px 10px 17px 14px;

  border-bottom: 1px solid ${theme.color.ghost};

  cursor: pointer;

  &:first-child {
      border-top: 1px solid ${theme.color.blackPearl};
    }

  &:last-child {
      border: none;
    }
`;

Calculator.InputContainer = styled.div`
  position: relative;

  margin-bottom: 11px;

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    margin-bottom: 1px;
  }

  ${(props) => {
    switch (props.$type) {
      case InputTypes.initialFee:
        return css`
          margin-bottom: 0;

          @media (min-width: ${theme.tabletWidthMinThreshold}) {
            margin-bottom: 0;
          }
        `;

      case InputTypes.term:
        return css`
          margin-bottom: 3px;

          @media (min-width: ${theme.tabletWidthMinThreshold}) {
            margin-bottom: 3px;
          }
        `;

      case InputTypes.userInfo:
        return css`
          margin-bottom: 30px;
          padding-right: 3px;
          padding-left: 3px;

          @media (min-width: ${theme.tabletWidthMinThreshold}) {
            display: flex;
            flex-wrap: wrap;
            justify-content: space-between;

            margin-bottom: 20px;
          }

          @media (min-width: ${theme.desktopWidthMinThreshold}) {
            padding-right: 70px;
            padding-left: 70px;
          }
        `;

      default:
        return css`
        `;
    }
  }}
`;

Calculator.Input = styled.input`
  ${input()};
`;

Calculator.RegApplication = styled.form`
  padding-top: 43px;

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    padding-top: 55px;
  }

  @media (min-width: ${theme.desktopWidthMinThreshold}) {
    margin-top: 37px;
    margin-right: auto;
    margin-left: auto;
    padding-top: 63px;
    padding-bottom: 70px;

    width: 770px;

    background-color: ${theme.color.ghostWhite};
  }
`;

Calculator.RequestTable = styled.table`
  width: 100%;

  margin-right: 0;
  margin-bottom: 13px;
  padding: 1px 1px 1px 1px;

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    margin-bottom: 15px;
    padding: 18px 1px 1px 1px;
  }

  @media (min-width: ${theme.desktopWidthMinThreshold}) {
    padding: 4px 67px 1px 67px;
  }
`;

Calculator.RequestField = styled.tr`
  display: block;

  width: 100%;

  margin-bottom: 14px;
  padding-bottom: 10px;

  border-bottom: 1px solid ${theme.color.silver};

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    display: flex;
    flex-direction: row-reverse;
    justify-content: space-between;

    margin-bottom: 12px;
    padding-bottom: 9px;
  }
`;

Calculator.RequestValue = styled.td`
  display: block;

  font-weight: 500;
  font-size: 18px;
  line-height: 25px;

  color: ${theme.color.jaguar};

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    font-size: 22px;
    line-height: 31px;
  }
`;

Calculator.RequestName = styled.td`
  display: block;

  font-size: 16px;
  line-height: 22px;

  color: ${theme.color.cello};

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    padding-top: 5px;
  }
`;

Calculator.SubmitButton = styled.button`
  ${submitButton()};
`;

Calculator.Popup = styled.div`
  position: fixed;

  display: flex;
  justify-content: center;
  align-items: center;

  width: 100vw;
  height: 100vh;

  top: 0;
  left: 0;

  background-color: rgba(88, 87, 87, 0.6);

  z-index: 1;
`;

Calculator.PopupContainer = styled.div`
  position: fixed;

  width: 290px;
  min-height: 198px;

  padding: 35px 25px 27px 25px;

  background-color: ${theme.color.ghostWhite};
  border-radius: 4px;
  box-sizing: border-box;

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    width: 678px;
    min-height: 193px;

    padding-top: 52px;
    padding-bottom: 15px;
  }

  @media (min-width: ${theme.desktopWidthMinThreshold}) {
    width: 500px;
  }
`;

Calculator.PopupClose = styled.button`
  position: absolute;
  display: block;

  width: 15px;
  height: 15px;

  top: 14px;
  right: 14px;

  background-image: url(${iconPopupClose});
  background-repeat: no-repeat;
  background-color: ${theme.color.ghostWhite};
  background-size: cover;

  border: none;
  cursor: pointer;

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    top: 30px;
    right: 30px;
  }

  @media (min-width: ${theme.desktopWidthMinThreshold}) {
    width: 18px;
    height: 18px;

    top: 25px;
    right: 25px;
  }
`;

Calculator.PopupTitle = styled.h2`
  width: 200px;

  margin: 0 auto 13px;

  font-size: 18px;
  line-height: 25px;
  text-align: center;

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    font-weight: 500;
    font-size: 22px;
    line-height: 31px;

    text-align: center;

    width: 400px;

    margin: 0 auto 13px;
  }
`;

Calculator.PopupContent = styled.p`
  width: auto;

  font-size: 16px;
  line-height: 22px;
  text-align: center;

  margin: 0 auto;

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    width: 300px;
  }
`;

export default Calculator;
