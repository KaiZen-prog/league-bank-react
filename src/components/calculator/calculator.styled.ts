import styled, {StyledComponentBase} from 'styled-components';
import theme from '../../theme/theme';

interface ICalculator extends StyledComponentBase<any, object> {}

const Calculator: ICalculator = styled.section`
  padding: 53px 20px 10px;

  margin: auto;

  @media (min-width: ${theme.tabletWidthMinThreshold}) {
    padding: 69px 42px 40px;
  }

  @media (min-width: ${theme.desktopWidthMinThreshold}) {
    padding: 96px 20px 40px 20px;

    max-width: 1210px;
    margin: auto;
  }
`;

export default Calculator;
