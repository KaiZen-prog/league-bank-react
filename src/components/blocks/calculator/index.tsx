import React, {useEffect} from 'react';
import {useAppSelector, useAppDispatch} from '../../../hooks/hooks';
import {ActionType} from '../../../store/actions/calculator';
import Section from '../../UI/section/section';
import ModalConfirm from '../../modals/modal-confirm';
import CalculatorForm from '../calculator-form';
import ApplicationForm from '../applicaion-form';

const Calculator: React.FunctionComponent = () => {
  const state = useAppSelector((store) => store.calculator);
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch({type: ActionType.SET_CREDIT_DATA});
  }, [
    state.initialFee,
    state.term,
    state.purpose,
    state.cost,
    state.casco,
    state.lifeInsurance,
    state.maternalCapital,
  ]);

  return (
    <Section>
      <CalculatorForm/>
      {state.step >= 3 && (<ApplicationForm/>)}
      {state.step >= 4 && (<ModalConfirm/>)}
    </Section>
  );
};

Calculator.displayName = 'Calculator';

export default Calculator;
