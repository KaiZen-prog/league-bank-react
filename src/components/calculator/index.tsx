import React, {useEffect} from 'react';
import {useAppSelector, useAppDispatch} from '../../hooks/hooks';
import {ActionType} from '../../store/actions/calculator';
import Block from './calculator.styled';
import PopupConfirm from '../popup-confirm';
import CalculatorForm from '../calculator-form';
import ApplicationForm from '../applicaion-form';

const Calculator: React.FunctionComponent = () => {
  const state = useAppSelector((store) => store.calculator);
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch({type: ActionType.SET_CREDIT_DATA});
  }, [
    state.initialFee,
    state.term,
    state.purpose,
    state.cost,
    state.casco,
    state.lifeInsurance,
    state.maternalCapital,
  ]);

  return (
    <Block>
      <CalculatorForm/>
      {state.step >= 3 && (<ApplicationForm/>)}
      {state.step >= 4 && (<PopupConfirm/>)}
    </Block>
  );
}

Calculator.displayName = 'Calculator';

export default Calculator;
