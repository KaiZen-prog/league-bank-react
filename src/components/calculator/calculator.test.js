import React, {createRef} from 'react';
import renderer from "react-test-renderer";
import {Provider} from 'react-redux';
import {Router} from "react-router-dom";
import {createMemoryHistory} from "history";
import configureStore from 'redux-mock-store';
import {Calculator} from "./calculator";
import {InitialState} from "../../mocks/test-mocks";

const mockStore = configureStore();
const history = createMemoryHistory({initialEntries: [`/`]});

const noop = () => {};

describe(`Calculator render correctly`, () => {
  const store = mockStore(InitialState);

  it(`Calculator step 1 closed`, () =>{
    let state = {
      step: 1,
      purpose: `none`,
      isPurposeSelectOpened: false,
      paramsCredit: {},

      cost: 0,
      initialFee: 0,
      term: 0,

      maternalCapital: false,
      casco: false,
      lifeInsurance: false,

      creditAmount: 0,
      percent: `0`,
      monthlyPayment: 0,
      requiredIncome: 0,

      isLabelClicked: false,
      isFormValid: true
    };

    const tree = renderer
        .create(
            <Provider store={store}>
              <Router history={history}>
                <Calculator
                  costInputRef={createRef()}
                  costDivRef={createRef()}
                  initialFeeInputRef={createRef()}
                  initialFeeDivRef={createRef()}
                  termInputRef={createRef()}
                  termDivRef={createRef()}
                  telRef={createRef()}
                  state={state}
                  onSelectOpen={noop}
                  onSelectClose={noop}
                  onPurposeChange={noop}
                  onLabelClick={noop}
                  onInputFocus={noop}
                  onInputChange={noop}
                  onCostChange={noop}
                  onInitialFeeChange={noop}
                  onTermChange={noop}
                  onInputRangeChange={noop}
                  onAdditionalChange={noop}
                  onCostChangeSign={noop}
                  onMakeRequest={noop}
                  onSubmit={noop}
                  onPopupClose={noop}
                  onRegApplicationChange={noop}
                  onChangePhone={noop}
                  requestNumber={1}
                />
              </Router>
            </Provider>
        ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Calculator step 1 opened`, () =>{
    let state = {
      step: 1,
      purpose: `none`,
      isPurposeSelectOpened: true,
      paramsCredit: {},

      cost: 0,
      initialFee: 0,
      term: 0,

      maternalCapital: false,
      casco: false,
      lifeInsurance: false,

      creditAmount: 0,
      percent: `0`,
      monthlyPayment: 0,
      requiredIncome: 0,

      isLabelClicked: false,
      isFormValid: true
    };

    const tree = renderer
        .create(
            <Provider store={store}>
              <Router history={history}>
                <Calculator
                  costInputRef={createRef()}
                  costDivRef={createRef()}
                  initialFeeInputRef={createRef()}
                  initialFeeDivRef={createRef()}
                  termInputRef={createRef()}
                  termDivRef={createRef()}
                  telRef={createRef()}
                  state={state}
                  onSelectOpen={noop}
                  onSelectClose={noop}
                  onPurposeChange={noop}
                  onLabelClick={noop}
                  onInputFocus={noop}
                  onInputChange={noop}
                  onCostChange={noop}
                  onInitialFeeChange={noop}
                  onTermChange={noop}
                  onInputRangeChange={noop}
                  onAdditionalChange={noop}
                  onCostChangeSign={noop}
                  onMakeRequest={noop}
                  onSubmit={noop}
                  onPopupClose={noop}
                  onRegApplicationChange={noop}
                  onChangePhone={noop}
                  requestNumber={1}
                />
              </Router>
            </Provider>
        ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
