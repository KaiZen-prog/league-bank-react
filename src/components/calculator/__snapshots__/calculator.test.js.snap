// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator render correctly Calculator step 1 closed 1`] = `
<section
  className="calculator"
>
  <a
    name="calculator"
  />
  <form
    action="#"
    className="calculator__form"
    onSubmit={[Function]}
  >
    <h2
      className="calculator__title"
    >
      Кредитный калькулятор
    </h2>
    <div
      className="calculator__flex-container"
    >
      <div
        className="calculator__container"
      >
        <fieldset
          className="calculator__purpose"
        >
          <h3
            className="calculator__step"
          >
            Шаг 1. Цель кредита
          </h3>
          <div
            className="calculator__purpose-select"
            onClick={[Function]}
          >
            <span
              className="calculator__select-title"
            >
              Выберите цель кредита
            </span>
            <ul
              className="calculator__purpose-list calculator__purpose-list--hidden"
            >
              <li
                className="calculator__purpose-item"
                id="mortgage"
                onClick={[Function]}
              >
                Ипотечное кредитование
              </li>
              <li
                className="calculator__purpose-item"
                id="car"
                onClick={[Function]}
              >
                Автомобильное кредитование
              </li>
            </ul>
          </div>
        </fieldset>
      </div>
    </div>
  </form>
</section>
`;

exports[`Calculator render correctly Calculator step 1 opened 1`] = `
<section
  className="calculator"
>
  <a
    name="calculator"
  />
  <form
    action="#"
    className="calculator__form"
    onSubmit={[Function]}
  >
    <h2
      className="calculator__title"
    >
      Кредитный калькулятор
    </h2>
    <div
      className="calculator__flex-container"
    >
      <div
        className="calculator__container"
      >
        <fieldset
          className="calculator__purpose"
        >
          <h3
            className="calculator__step"
          >
            Шаг 1. Цель кредита
          </h3>
          <div
            className="calculator__purpose-select calculator__purpose-select--opened"
            onClick={[Function]}
          >
            <span
              className="calculator__select-title"
            >
              Выберите цель кредита
            </span>
            <ul
              className="calculator__purpose-list"
            >
              <li
                className="calculator__purpose-item"
                id="mortgage"
                onClick={[Function]}
              >
                Ипотечное кредитование
              </li>
              <li
                className="calculator__purpose-item"
                id="car"
                onClick={[Function]}
              >
                Автомобильное кредитование
              </li>
            </ul>
          </div>
        </fieldset>
      </div>
    </div>
  </form>
</section>
`;
